YamlUtility/src/main/java/my/com/midf/digitalbusiness/devops/jasonfyp/YamlUtilityImpl.java

```diff
@@ -35,86 +34,68 @@ public class YamlUtilityImpl implements YamlUtility {
-            }
+		} catch (FileNotFoundException ex) {
+			Logger.getLogger(YamlUtilityImpl.class.getName()).log(Level.SEVERE, null, ex);
+		} finally {
+			try {
+				is.close();
+			} catch (IOException ex) {
+				Logger.getLogger(YamlUtilityImpl.class.getName()).log(Level.SEVERE, null, ex);
+			}
+		}
 	}
 
 	/**
 	 * extract an object from content field above by mentioning the key
 	 */
 	@Override
 	public String extract(String key) {
-            String[] keyArray = key.split("\\.");
-            return recursiveExtract(0, keyArray, content);
-        }
-        
-        private String recursiveExtract(int index, String[] keyArray, Map map){
-            try{
-                String stringToExtract = "";
-                String key = keyArray[index];
-                if(doesThisMapHas(key, map)){
-                    String newContent = ""+map.get(key);
-                    if(isThisAYaml(newContent)){
-                        Yaml yaml = new Yaml();
-                        newContent = newContent.replaceFirst("\\{", "");
-                        newContent = newContent.substring(0, newContent.length()-1);
-                        System.out.println(newContent);
-                        
-                        Map<String, Object> newMap =  yaml.load(newContent);
-                        index++;
-                        recursiveExtract(index, keyArray, newMap);
-                    }
-                    stringToExtract = newContent;
-                }
-                else{
-                    System.out.println("key "+key+" does not exist");
-                }
-                return stringToExtract;
-            }
-            catch(Exception ex){
-                ex.printStackTrace();
-            }
-            return "";
-        }
-        
-        private boolean isThisAYaml(String possibleYamlString){
+		String[] keyArray = key.split("\\.");
+		String extractedValue = recursiveExtract(0, keyArray, content);
+		return extractedValue;
+	}
 
-            if(possibleYamlString.contains("{")){
-                return true;
-            }
-            else return false;
-        }
-        
-        private boolean doesThisMapHas(String key, Map map){
-            return map.containsKey(key);
-        }
-        
-        private String getValue(String key, Map map){
-            return map.get(key)+"";
-        }
-        
-        private Map getSubMap(String key, Map map){
-            Map<String, Object> smallerMap;
-           Object newContent = map.get(key);
-           Yaml yaml = new Yaml();
-           smallerMap = yaml.load(newContent+"");
-            return smallerMap;
-        }
+	private String recursiveExtract(int index, String[] keyArray, Map map) {
+		try {
+			String stringToExtract = "";
+			String key = keyArray[index];
+			if (doesThisMapHas(key, map)) {
+				if (isThisAYaml("" + map.get(key))) {
+					Yaml yaml = new Yaml();
+
+					Map<String, Object> newMap = yaml.load(yaml.dump(map.get(key)));
```

Comment:
the trick is to use yaml.dump to convert an object into yaml string, then load them back into map object.